### PR TITLE
[Transforms] Fixes issues that were causing runtests-browser to fail

### PR DESCRIPTION
**Pull request by rbuckton**
Even with this fix, transforms are still not enabled by default when running tests in the browser, but this fixes a few issues that were blocking runtests-browser from running at all.

This also adds support for `getEnvironmentVariable` with ChakraHost..

### CODE REVIEW

**Comment by rbuckton** [comment]
CC: @vladima, @yuit

File: src/harness/harness.ts [hunk 1]
```diff
@@ -1681,7 +1681,7 @@ namespace Harness {
     if (Error) (<any>Error).stackTraceLimit = 1;
 }
 
-if (ts.sys.tryEnableSourceMapsForHost && /^development$/i.test(ts.sys.getEnvironmentVariable("NODE_ENV"))) {
+if (ts.sys && ts.sys.tryEnableSourceMapsForHost && /^development$/i.test(ts.sys.getEnvironmentVariable("NODE_ENV"))) {
```
Comment (mhegazy) [inline src/harness/harness.ts]: this should be on `IO` instead

Comment (rbuckton) [inline src/harness/harness.ts]: I've just pushed up a commit that adds entries for these on IO.

File: src/compiler/program.ts [hunk 2]
```diff
@@ -644,7 +644,8 @@ namespace ts {
             fileExists: fileName => sys.fileExists(fileName),
             readFile: fileName => sys.readFile(fileName),
             trace: (s: string) => sys.write(s + newLine),
-            directoryExists: directoryName => sys.directoryExists(directoryName)
+            directoryExists: directoryName => sys.directoryExists(directoryName),
+            getEnvironmentVariable: sys.getEnvironmentVariable
```
Comment (vladima) [inline src/compiler/program.ts]: can you wrap it into lambda so `this` won't be torn out?

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**